python-checks/src/main/java/org/sonar/python/checks/OverwrittenCollectionEntryCheck.java

```diff
@@ -121,6 +123,25 @@ private static CollectionWrite collectionWrite(AssignmentStatement assignment, E
     return null;
   }
 
+  @CheckForNull
+  private static String key(SubscriptionExpression subscriptionExpression) {
```

Comment:
I'm confused that we have to recompute the key part of the subscription here.

As I understand the rule, in `reportOverwrites` we should check for a given key if the writes are made on the same collection. Isn't there where the bug is? It seems that in this method if the 2 collections (not key) are different, we should simply not raise the issue.